R/cast.R

```diff
@@ -0,0 +1,45 @@
+#' Casting
+#'
+#' Double dispatch methods to support [vctrs::vec_cast()].
+#'
+#' @inheritParams vctrs::vec_cast
+#'
+#' @method vec_cast blob
+#' @export
+#' @export vec_cast.blob
+vec_cast.blob <- function(x, to) UseMethod("vec_cast.blob")
+
+#' @method vec_cast.blob default
+#' @export
+vec_cast.blob.default <- function(x, to) stop_incompatible_cast(x, to)
+
+#' @method vec_cast.blob logical
+#' @export
+vec_cast.blob.logical <- function(x, to) vec_unspecified_cast(x, to)
+
+#' @method vec_cast.blob blob
+#' @export
+vec_cast.blob.blob <- function(x, to) x
+
+#' @method vec_cast.blob list
+#' @export
+vec_cast.blob.list <- function(x, to) blob(!!!x)
```

Comment:
Why doesn't this use `vec_list_cast()`?

Reply:
`vec_list_cast()` discards all elements after the first, we want to treat a list as a list-of-raw. Or do we?

Reply:
Ah got it.